src/rpc/misc.cpp

```diff
@@ -188,8 +196,11 @@ UniValue validateaddress(const JSONRPCRequest& request)
         }
         if (pwallet) {
             const auto& meta = pwallet->mapKeyMetadata;
-            const CKeyID *keyID = boost::get<CKeyID>(&dest);
-            auto it = keyID ? meta.find(*keyID) : meta.end();
+            CKeyID key = GetKeyForDestination(*pwallet, dest);
```

Comment:
Below you use `auto key = ...`. Keep it consistent?

Reply:
Choosing to use `auto` is a balance between clarity (by being explicit) and brevity/flexibility. I don't think using it here would be an improvement, but I added it on the line below (the iterator type was pretty long...).